configuration_test.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+import sys
+import os
+import re
+import subprocess
+import pkg_resources
+import warnings
+
+def configuration_test(ARGUMENTS):
+    # Determines whether to print traceback messages
+    debug = ARGUMENTS.get('debug', False) 
+    if debug == False:
+        # Hide traceback for configuration test only
+        # http://stackoverflow.com/questions/27674602/hide-traceback-unless-a-debug-flag-is-set
+        sys.tracebacklimit = 0 
+
+    # Checks initial prerequisites
+    check_python()
+    check_lyx()
+    check_gitlfs()
+    check_yamls()
+
+    # Uncomment if using
+    #check_r() 
+    #check_gslab_metropolis() 
+
+    # Loads arguments and configurations
+    mode = ARGUMENTS.get('mode', 'develop') # Gets mode; defaults to 'develop'
+
+    # Check stata
+    sf = check_stata(ARGUMENTS)
+
+    # Checks mode/version
+    if not (mode in ['develop', 'cache']):
+        raise PrerequisiteError("Error: %s is not a defined mode" % mode)
+    
+    # Get return list
+    if mode == 'cache':
+        cache_dir    = os.path.expanduser(load_yaml_value("user-config.yaml", "cache"))
+        check_cache(cache_dir)
+        return_list = [mode, sf, cache_dir]
+    else:
+        return_list = [mode, sf, None]
+
+    # Restore default tracebacklimit and return values
+    sys.tracebacklimit = 0
+    return return_list
+
+def check_python():
+    if sys.version_info[0] != 2:
+        raise PrerequisiteError('Please use python 2')
+    check_python_packages()
+
+def check_python_packages():
+    try:
+        import gslab_scons
+        import gslab_make
+        import gslab_fill
+    except:
+        raise PrerequisiteError('Missing gslab_python modules')
+
+    try:
+        import yaml
+    except:
+        raise PrerequisiteError('Missing PyYAML python module')
+
+    if pkg_resources.get_distribution('gslab_tools').version < '3.0.3':
+        raise PrerequisiteError('Wrong version of gslab_python modules')
+
+def check_r():
```

Comment:
Perhaps `gslab_scons` is a better home for these functions than this template script? This could reduce the amount of code we duplicate across the repos and expedite the process of making universal changes.